Std/Data/BitVec/Basic.lean

```diff
@@ -316,9 +310,18 @@ Bitwise NOT for bit vectors.
 ```
 SMT-Lib name: `bvnot`.
 -/
-protected def not (x : BitVec n) : BitVec n := -(x + .ofNat n 1)
+protected def not (x : BitVec n) : BitVec n :=
+  .ofFin ⟨(1 <<< n).pred,
+    n.one_shiftLeft.symm ▸
+      (Nat.pred_lt <| Nat.ne_of_gt <| Nat.pos_pow_of_pos _ <| Nat.zero_lt_succ _)⟩ ^^^ x
```

Comment:
Seems like the first term could be made into a separate definition -- maybe `ones : BitVec n`?

Reply:
My hesitation there is that we now have three ways to spell the vector of all ones; `-1`, `~~~0`, and `ones`.

Reply:
That seems ok to me.  It has good algebraic properties and couldn't be expressed with the direct property you wanted in the code above.

Another possibility would be to also modify `neg` (and introduce `one`) so that `-1` could be used in place of the code above.